app/src/main/java/com/example/tasky/common/presentation/TextBox.kt

```diff
@@ -14,32 +16,76 @@ import androidx.compose.runtime.remember
 import androidx.compose.runtime.setValue
 import androidx.compose.ui.Modifier
 import androidx.compose.ui.graphics.Color
+import androidx.compose.ui.res.colorResource
+import androidx.compose.ui.res.painterResource
+import androidx.compose.ui.text.input.PasswordVisualTransformation
+import androidx.compose.ui.text.input.VisualTransformation
 import androidx.compose.ui.tooling.preview.Preview
 import androidx.compose.ui.unit.dp
+import com.example.tasky.R
+import com.example.tasky.common.domain.isValidName
 
 @Composable
-fun TextBox(hintText: String) {
-    var text by remember { mutableStateOf("") }
+fun TextBox(
+    hintText: String,
+    text: String,
+    onValueChange: (String) -> Unit,
+    validator: (String) -> Boolean,
+    isPasswordField: Boolean = false
+) {
+    var isValid by remember { mutableStateOf(validator(text)) }
+    var passwordVisible by remember { mutableStateOf(false) }
```

Comment:
Same here, such logic should be put at a central place in your ViewModel. Not only will that make it safe against configuration changes like screen rotations (states with `remember` will be reset after a config change), but you'll also have clear responsibilities for both the pure UI and the logic that decides how the UI should look like. 

As mentioned earlier, that will make it easier to test the UI logic, since for testing a ViewModel you don't need to spin up a real Android device and simulate UI, but rather call some simple functions.

In general, UI should always stay as dumb as possible. So rather than letting the UI decide what a valid password/email is, the ViewModel should decide that and just expose a state like `isEmailValid` which the UI can then decide what that means for the screen.